addons/report/models/res_company.py

```diff
@@ -22,9 +22,12 @@ def set_report_template(self):
         self.ensure_one()
         if self.env.context.get('report_template', False):
             self.external_report_layout = self.env.context['report_template']
-        if self.env.context.get('default_report_name'):
-            document = self.env.get(self.env.context['active_model']).browse(self.env.context['active_id'])
-            return self.env['report'].get_action(document, self.env.context['default_report_name'], config=False)
+        report = self.env.context.get('default_report_name')
+        if report:
+            return {
+                'type': 'ir.actions.report.xml',
+                'report_name': report,
+            }
```

Comment:
I don't understand why this was changed, it doesn't have the same behavior as before.

While testing this, I get: https://drive.google.com/a/odoo.com/file/d/0Bw8vRpv1R7NKdm5FTHdMU3k2RGc/view?usp=drivesdk now,

while I got https://drive.google.com/a/odoo.com/file/d/0Bw8vRpv1R7NKbVRzTXNMMEw5aVE/view?usp=drivesdk before.